lib/rules/role-supports-aria-props.js

```diff
@@ -0,0 +1,101 @@
+// @ts-check
+const {aria, elementRoles, roles} = require('aria-query')
+const {getProp, getPropValue, propName} = require('jsx-ast-utils')
+const {getElementType} = require('../utils/get-element-type')
+const ObjectMap = require('../utils/object-map')
+
+// Clean-up `elementRoles` from `aria-query`
+const elementRolesMap = new ObjectMap()
+for (const [key, value] of elementRoles.entries()) {
+  // - Remove unused `constraints` key
+  delete key.constraints
+  key.attributes = key.attributes?.filter(attribute => !('constraints' in attribute))
+  // - Remove empty `attributes` key
+  if (!key.attributes || key.attributes?.length === 0) {
+    delete key.attributes
+  }
+  elementRolesMap.set(key, value)
+}
+// - Remove insufficiently-disambiguated `menuitem` entry
+elementRolesMap.delete({name: 'menuitem'})
+// - Disambiguate `menuitem` and `menu` roles by `type`
+elementRolesMap.set({name: 'menuitem', attributes: [{name: 'type', value: 'command'}]}, ['menuitem'])
+elementRolesMap.set({name: 'menuitem', attributes: [{name: 'type', value: 'radio'}]}, ['menuitemradio'])
+elementRolesMap.set({name: 'menuitem', attributes: [{name: 'type', value: 'toolbar'}]}, ['toolbar'])
+elementRolesMap.set({name: 'menu', attributes: [{name: 'type', value: 'toolbar'}]}, ['toolbar'])
+
+module.exports = {
+  meta: {
+    docs: {
+      description:
+        'Enforce that elements with explicit or implicit roles defined contain only `aria-*` properties supported by that `role`.',
+      url: require('../url')(module)
+    },
+    schema: []
+  },
+
+  create(context) {
+    return {
+      JSXOpeningElement(node) {
+        // Assemble a key for looking-up the element’s role in the `elementRolesMap`
+        // - Get the element’s name
+        const key = {name: getElementType(context, node)}
+        // - Get the element’s disambiguating attributes
+        for (const prop of ['aria-expanded', 'type', 'size', 'role', 'href', 'multiple', 'scope']) {
```

Comment:
Originally, I’d dynamically generated this list of the attributes that are necessary for disambiguating elements with multiple roles (e.g. `<input type="radio">` vs `<input type="checkbox">`). But, given [my nonchalance towards separation of concerns](https://github.com/github/eslint-plugin-github/pull/362/files#r1051327282), the readability and performance gains, and the fact that this can’t lead to false positives, I opted for this simpler hardcoded list.